### PR TITLE
nsnotification center stuff

### PR DESCRIPTION
**Pull request by sameerav**

### CODE REVIEW

File: sage-ios/SAGE/SAGE/AddAnnouncementController.swift [hunk 1]
```diff
@@ -40,6 +40,7 @@ class AddAnnouncementController: UIViewController {
             let finalAnnouncement = self.addAnnouncementView.exportToAnnouncement()
             AdminOperations.createAnnouncement(finalAnnouncement, completion: { (announcement) -> Void in
                 self.navigationController!.popViewControllerAnimated(true)
+                NSNotificationCenter.defaultCenter().postNotificationName(NotificationConstants.addAnnouncementKey, object: finalAnnouncement)
```
Comment (amillman) [inline sage-ios/SAGE/SAGE/AddAnnouncementController.swift]: finalAnnouncement or the announcement passed in the completion block? probably better to use the anonuncement instead since it retains the correct id number

File: sage-ios/SAGE/SAGE/AnnouncementsViewController.swift [hunk 2]
```diff
@@ -14,6 +14,41 @@ class AnnouncementsViewController: UITableViewController {
...
+    
+    deinit {
+        NSNotificationCenter.defaultCenter().removeObserver(self)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func addAnnouncement(notification: NSNotification) {
```
Comment (amillman) [inline sage-ios/SAGE/SAGE/AnnouncementsViewController.swift]: can you rename to like announcementAdded instead? or else it sounds like an action done by this controller

File: sage-ios/SAGE/SAGE/AnnouncementsViewController.swift [hunk 3]
```diff
@@ -14,6 +14,41 @@ class AnnouncementsViewController: UITableViewController {
...
+        let announcement = notification.object as! Announcement
+        self.announcements.insert(announcement, atIndex: 0)
+        let indexPath = NSIndexPath(forRow: 0, inSection: 0)
+        self.tableView.insertRowsAtIndexPaths([indexPath], withRowAnimation: .Automatic)
+    }
+    
+    func editAnnouncement(notification: NSNotification) {
```
Comment (amillman) [inline sage-ios/SAGE/SAGE/AnnouncementsViewController.swift]: same here. naming

File: sage-ios/SAGE/SAGE/AnnouncementsViewController.swift [hunk 4]
```diff
@@ -34,6 +69,8 @@ class AnnouncementsViewController: UITableViewController {
         self.getAnnouncements()
     }
     
+    // MARK: - NSNotificationCenter selectors
```
Comment (amillman) [inline sage-ios/SAGE/SAGE/AnnouncementsViewController.swift:76]: put a // belo and above the mark line

File: sage-ios/SAGE/SAGE/BrowseSchoolsViewController.swift [hunk 5]
```diff
@@ -14,6 +14,45 @@ class BrowseSchoolsViewController: UITableViewController {
...
+        NSNotificationCenter.defaultCenter().removeObserver(self)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    // MARK: - NSNotificationCenter selectors
+
+    func addSchool(notification: NSNotification) {
```
Comment (amillman) [inline sage-ios/SAGE/SAGE/BrowseSchoolsViewController.swift]: same

File: sage-ios/SAGE/SAGE/AdminTableViewController.swift [hunk 6]
```diff
@@ -52,7 +52,7 @@ class AdminTableViewController: UITableViewController {
             case 0:
                 self.navigationController!.pushViewController(BrowseMentorsViewController(), animated: true)
             case 1:
-                self.navigationController!.pushViewController(BrowseSchoolsViewController(), animated: true)
+                self.navigationController!.pushViewController(BrowseSchoolsViewController(style: .Plain), animated: true)
```
Comment (amillman) [inline sage-ios/SAGE/SAGE/AdminTableViewController.swift:55]: use ?

Comment (sameerav) [inline sage-ios/SAGE/SAGE/AdminTableViewController.swift:55]: will do in separate pr

File: sage-ios/SAGE/SAGE/AnnouncementsViewController.swift (hunk 4 above)
Comment (amillman) [inline sage-ios/SAGE/SAGE/AnnouncementsViewController.swift:76]: ples
